components/permissions/contexts/brave_ethereum_permission_context.cc

```diff
@@ -71,4 +93,76 @@ void BraveEthereumPermissionContext::RequestPermission(
                                            user_gesture, std::move(callback));
 }
 
+// static
+void BraveEthereumPermissionContext::Accept(
+    const std::vector<std::string>& accounts,
+    content::WebContents* web_contents) {
+  PermissionRequestManager* manager =
+      PermissionRequestManager::FromWebContents(web_contents);
+  DCHECK(manager);
+
+  std::vector<PermissionRequest*> allowed_requests;
+  std::vector<PermissionRequest*> cancelled_requests;
+  for (PermissionRequest* request : manager->Requests()) {
+    if (IsAccepted(request, accounts)) {
+      allowed_requests.push_back(request);
+    } else {
+      cancelled_requests.push_back(request);
+    }
+  }
+
+  manager->AllowOrCancel(allowed_requests, cancelled_requests);
```

Comment:
having `AllowOrCancel` still seems specific to ethereum. I think it would be better to have separate `Allow`, `Cancel` and `Deny` methods.

Reply:
also please change references `allow` -> `accept` in general to keep standard permission terminology

Reply:
https://github.com/brave/brave-core/commit/39529121ec8feba7ceb823c6ba12861e24d5c6ff